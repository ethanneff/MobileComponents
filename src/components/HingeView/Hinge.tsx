import React from "react";
import { Circle } from "react-native-svg";
import { useBoneMode } from "../../commons/Hooks/useBoneMode";
import { Point2 } from "../../commons/Utils/Maths/geometry/Point";

interface HingeProps {
  position: Point2;
  id?: string;
  strokeColor: string;
  strokeWidth?: number;
  fillColor: string;
  radius: number;
}

export default function Hinge({
  strokeWidth,
  strokeColor,
  fillColor,
  radius,
  position,
  id
}: HingeProps) {
  const boneMode = useBoneMode();
  if (boneMode) {
    return null;
  }
  return (
    <Circle
      id={id}
      cx={position.x}
      cy={position.y}
      r={radius}
      stroke={strokeColor}
      strokeWidth={strokeWidth}
      fill={fillColor}
    />
  );
}
