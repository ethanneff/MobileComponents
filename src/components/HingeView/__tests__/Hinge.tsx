import React from "react";
import Victor from "victor";
import { testRender } from "../../../commons/Utils/TestMocks/test-render";
import Hinge from "../Hinge";

describe("Hinge Component", () => {
  it("renders correctly", () => {
    const position = new Victor(10, 20);
    const { tree } = testRender(
      <Hinge
        id="hinge"
        position={position}
        radius={15}
        strokeWidth={10}
        strokeColor={"green"}
        fillColor={"black"}
      />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
