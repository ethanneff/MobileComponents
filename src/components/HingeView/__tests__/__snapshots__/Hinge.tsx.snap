// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hinge Component renders correctly 1`] = `
<RNSVGCircle
  cx={10}
  cy={20}
  fill={4278190080}
  fillOpacity={1}
  fillRule={1}
  matrix={
    Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ]
  }
  name="hinge"
  opacity={1}
  originX={0}
  originY={0}
  propList={
    Array [
      "fill",
      "stroke",
      "strokeWidth",
    ]
  }
  r={15}
  rotation={0}
  scaleX={1}
  scaleY={1}
  skewX={0}
  skewY={0}
  stroke={4278222848}
  strokeDasharray={null}
  strokeDashoffset={null}
  strokeLinecap={0}
  strokeLinejoin={0}
  strokeMiterlimit={4}
  strokeOpacity={1}
  strokeWidth={10}
  vectorEffect={0}
  x={0}
  y={0}
/>
`;
